src/main/java/com/singlestore/jdbc/client/column/StringColumn.java

```diff
@@ -113,13 +113,18 @@ public int getPrecision() {
   @Override
   public Object getDefaultText(final Configuration conf, ReadableByteBuf buf, MutableInt length)
       throws SQLDataException {
+    if (isBinary()) {
+      byte[] arr = new byte[length.get()];
+      buf.readBytes(arr);
+      return arr;
+    }
     return buf.readString(length.get());
   }
 
   @Override
   public Object getDefaultBinary(final Configuration conf, ReadableByteBuf buf, MutableInt length)
       throws SQLDataException {
-    return buf.readString(length.get());
+    return getDefaultText(conf, buf, length);
   }
```

Comment:
Hmm....
It looks weird to me that `getDefaultText` and `getDefaultBinary` return the same thing.
It is the same in all other codecs. Do you know what is the purpose of these two methods and why we don't return binary in `getDefaultBinary`?

Reply:
yes, the method call(getDefaultText/getDefaultBinary) is defined by protocol is used(text/binary) , isBinary flag defines column type, so we need this logic for both protocols